ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts

```diff
@@ -0,0 +1,114 @@
+import { UnitNoteDocTypes, UnitNoteStatusTypes } from '@/enums/unitNoteDocTypes'
+import { CancelUnitNoteIF, UnitNoteIF, UnitNotePanelIF } from '@/interfaces'
+import { CancellableUnitNoteTypes, NoticeOfCautionDropDown } from '@/resources'
+
+/**
+ * Contains logic specific to rendering the unit note panels
+*/
+export const useMhrUnitNotePanel = () => {
+  /**
+   * Adds details to unit notes based on cancel notes and remove cancelled notes
+   * @param unitNotes
+   * @returns UnitNoteIF
+   */
+
+  // Identify if a unit note is notice of caution or an extended/continued notice of caution
+  const isNoticeOfCautionOrRelatedDocType = (note: UnitNoteIF): boolean => {
+    return [UnitNoteDocTypes.NOTICE_OF_CAUTION,
+      UnitNoteDocTypes.CONTINUED_NOTE_OF_CAUTION,
+      UnitNoteDocTypes.EXTENSION_TO_NOTICE_OF_CAUTION].includes(note.documentType)
+  }
+
+  const handleCancelledUnitNotes = (unitNotes: UnitNoteIF[]): UnitNotePanelIF[] => {
+    // Uses the cancelledDocumentRegistrationNumber to map a unit note to its cancel note
+    const cancelUnitNotesMap = new Map<string, CancelUnitNoteIF>()
+
+    // Since notes are in order from most recent creation date and time,
+    // the cancel notes should be added to map before the corresponding note is encountered
+    unitNotes.forEach((note: UnitNotePanelIF) => {
+      if (note.documentType === UnitNoteDocTypes.NOTE_CANCELLATION) {
+        cancelUnitNotesMap.set((note as CancelUnitNoteIF).cancelledDocumentRegistrationNumber, note as CancelUnitNoteIF)
+      } else if (cancelUnitNotesMap.has(note.documentRegistrationNumber)) {
+        // Adds cancelledDateTime and overrides remarks and givingNoticeParty
+        const cancelNote = cancelUnitNotesMap.get(note.documentRegistrationNumber)
+        note.remarks = cancelNote.remarks
```

Comment:
Here not sure when remarks are replaced. Your PR description says it does overwrite the unit note, but I dont have the full context.

Reply:
The cancel note filling has a editable remarks filled that is prefilled with the note being cancelled remarks, what is in that field is shown in the panel has the note's remarks when the note is cancelled.

![image](https://github.com/bcgov/ppr/assets/77707952/f8288118-dc0d-4b90-ac3a-0140b2a265ee)
![image](https://github.com/bcgov/ppr/assets/77707952/682dbcc5-8fff-42f2-99b1-22b5019d0378)

Reply:
Thanks for clarification.